src/main/java/lotto/service/LottoService.java

```diff
@@ -0,0 +1,35 @@
+package lotto.service;
+
+import lotto.model.Lotto;
+import lotto.model.LottoData;
+
+import java.util.List;
+
+public class LottoService {
+    private LottoData lottoData;
+
+    public void init(List<List<Integer>> l, List<Integer> n, int bn) {
+        this.lottoData = LottoData.init(l, n, bn);
+    }
+
+    public int[] getResult() {
+        return lottoData.getResult();
+    }
+
+    public void matchNumber() {
+        List<Integer> intersectionList;
+        for (Lotto l : lottoData.getLottos()) {
+            intersectionList = l.getNumbers().stream().filter(n -> lottoData.getNumbers().contains(n)).toList();
+            if (intersectionList.size() < 3) continue;
+            if (intersectionList.size() == 6 && !isContainBonus(intersectionList)) {
+                lottoData.addPrize(4);
+                continue;
+            }
+            lottoData.addPrize(intersectionList.size() - 3);
+        }
+    }
+
```

Comment:
```suggestion
 public void matchNumber() {
    for (Lotto lotto : lottoData.getLottos()) {
        int matchCount = (int) lotto.getNumbers().stream()
                .filter(lottoData.getNumbers()::contains)
                .count();

        if (matchCount >= 3) {
            int index = (matchCount == 6 && !isContainBonus(lotto.getNumbers())) ? 4 : matchCount - 3;
            lottoData.addPrize(index);
        }
    }
}

```
여기 if문에서 .size()를 계속 중복호출하고있는것같아요 matchCount라는 변수에 담아서 사용하는것은 어떨까요??